File: src/Services/Catalog/Catalog.Domain/Entities/Course.cs
using Catalog.Domain.Entities.Enums;

namespace Catalog.Domain.Entities
{
    public record Course
    {
        public Guid Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public Guid CreatorId { get; set; }
        public string ShortDescription { get; set; } = string.Empty;
        public string LongDescription { get; set; } = string.Empty;
        public Language Language { get; set; }
        public Level Level { get; set; }
        public Subject Subject { get; set; }
        public int TimeToComplete { get; set; }
        public string Prerequisites { get; set; } = string.Empty;



        public List<Chapter> Chapters { get; set; } = new List<Chapter>();
        public List<Review> Reviews { get; set; } = new List<Review>();
        public List<string> SkillsToGain { get; set; } = new List<string>();
        public List<Faq> FAQs { get; set; } = new List<Faq>();
    }
}

File: src/Web/WebSPA/src/app/_models/course.ts
import { Chapter } from "./chapter";
import { Faq } from "./faq";
import { Review } from "./review";

export interface Course {
    id: string,
    title: string,
    shortDescription: string,
    longDescription: string,
    timeToComplete: number,
    language: 0,
    level: 0,
    subject: 0,
    chapters: Chapter[],
    prerequisites: string[],
    reviews: Review[],
    faqs: Faq[]
}

export enum Subject {
    "Web Development" = 0,
    "Data Science" = 1,
    "AI" = 2
}

export enum Language {
    "C#" = 0,
    "Java" = 1,
    "JavaScript" = 2,
    "HTML" = 3,
    "Python" = 4,
    "C++" = 5,
    "SQL" = 6
}

export enum Level {
    Beginner = 0,
    Intermediate = 1,
    Advanced = 2
}